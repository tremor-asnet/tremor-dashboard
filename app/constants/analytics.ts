/* Constants for Analytics Page */
export const SALE_STATISTICAL_TYPE = {
  REVENUE: "Revenue",
};
