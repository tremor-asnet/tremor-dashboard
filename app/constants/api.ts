import { ROUTER_API_URL } from "./routes";

export const API_ROUTES = {
  USERS: `${ROUTER_API_URL}/users`,
};
