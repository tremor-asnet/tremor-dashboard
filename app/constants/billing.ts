export const enum AGGREGATION_TYPE {
  SALARY = "Salary",
  PAYPAL = "Paypal",
}

export const enum AGGREGATION_DESCRIPTION {
  SALARY = "Belong Interactive",
  PAYPAL = "Freelance Payment",
}
