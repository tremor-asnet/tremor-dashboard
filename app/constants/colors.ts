type ColorChannelsChartType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
};

export const CHANNELS_CHART_COLOR: ColorChannelsChartType = {
  1: "blue-500",
  2: "red-500",
  3: "primary",
  4: "gray-500",
};

export const TRANSACTION_COLOR = {
  INCREASE: "few",
  DECREASE: "attention",
  PENDING: "secondary",
};
