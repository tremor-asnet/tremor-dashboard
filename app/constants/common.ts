export const EXCEPT_KEYS = {
  POSITIVE_INTEGER: ["e", "E", "+", "-", "."],
  POSITIVE_DOUBLE: ["e", "E", "+", "-"],
};

export const DIRECTION = {
  ASC: "asc",
  DESC: "desc",
};
