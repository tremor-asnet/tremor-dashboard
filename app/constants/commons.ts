export const ITEM_ACTION_PROJECT = [
  {
    key: "act",
    label: "Action",
  },
  {
    key: "anotherAct",
    label: "Another action",
  },
  {
    key: "somethingElse",
    label: "Something else here",
  },
];

export const ITEM_ACTION_SALES_DATE = [
  {
    key: "last7Days",
    label: "Last 7 days",
  },
  {
    key: "lastWeek",
    label: "Last week",
  },
  {
    key: "last30Days",
    label: "Last 30 days",
  },
];

export const URL_AVATAR_PROJECT = [
  {
    avatar: "/images/avatar/avatar-xs.webp",
  },
  {
    avatar: "/images/avatar/avatar-xs.webp",
  },
  {
    avatar: "/images/avatar/avatar-xs.webp",
  },
];

// Social link
export const SOCIAL_LINK = {
  facebook: "https://www.facebook.com/",
  twitter: "https://twitter.com/",
  instagram: "https://instagram.com",
};

export const REMEMBER_ME = "remember-me";

export const listOption: ListOptionProps[] = [
  { option: "Paid" },
  { option: "Refunded" },
  { option: "Canceled" },
];
