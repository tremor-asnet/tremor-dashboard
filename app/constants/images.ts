export const NOT_FOUND = {
  SRC: "/images/placeholder-image.png",
};
export const NOT_FOUND_IMAGE = "/images/no_image_square.jpg";
export const LOGO_SRC = {
  logo: "/assets/images/sidebar-logo.webp",
};
export const PLACEHOLDER_IMAGE = "/images/not-found-image.png";

export const AVATAR_SRC = {
  xs: "/images/avatar/avatar-xs.webp",
  sm: "/images/avatar/avatar-sm.webp",
  md: "/images/avatar/avatar-md.webp",
  lg: "/images/avatar/avatar-lg.webp",
};

export const CHART_SRC = {
  analytics: "/assets/images/analytics/analytics-sales-chart.webp",
};

export const METADATA = {
  HOME_PREVIEW: "/assets/images/home-preview.png",
  MASTERCARD: "/assets/images/mastercard.webp",
};

export const PRODUCT = {
  DETAIL: "/assets/images/products/product-details.jpg",
};

export const IMAGE_TO_PRINT = {
  LIGHT_MODE: "https://i.ibb.co/87R6BcQ/logo-td-dark.webp",
  DARK_MODE: "https://i.ibb.co/k8TVzJ1/logo-td-light.webp",
};
