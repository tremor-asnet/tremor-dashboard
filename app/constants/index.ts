export * from "./routes";
export * from "./regex";
export * from "./messages";
export * from "./profile";
export * from "./commons";
export * from "./flagSrc";
export * from "./images";
export * from "./sections";
export * from "./images";
export * from "./products";
export * from "./colors";
export * from "./units";
export * from "./sales";
export * from "./status";
export * from "./saleStatistical";
export * from "./analytics";
export * from "./breakpoints";
export * from "./billing";
export * from "./pincode";
export * from "./api";
