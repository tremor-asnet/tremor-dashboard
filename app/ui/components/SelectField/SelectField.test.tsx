import { render, queryAllByAttribute, fireEvent } from "@testing-library/react";

// Constants
import { TYPE_PRICE } from "@/constants";

// Components
import { SelectField } from "..";

describe("SelectField Component", () => {
  it("Should match snapshot", () => {
    const component = render(
      <SelectField label="Currency" options={TYPE_PRICE} />,
    );
    expect(component).toMatchSnapshot();
  });

  it.skip("renders correct number of options", () => {
    const getById = queryAllByAttribute.bind(null, "id");

    const { container } = render(
      <SelectField label="Currency" options={TYPE_PRICE} name="currency" />,
    );
    const selectElement = getById(container, "currency");
    expect(selectElement[0]?.children).toHaveLength(1);
  });

  it("renders correct name of options", () => {
    const { getAllByText } = render(
      <SelectField label="Currency" options={TYPE_PRICE} value="4" />,
    );
    expect(getAllByText("GBP")).toBeTruthy();
  });

  it("should call onChange handler with selected value", () => {
    const onChangeMock = jest.fn(); // Mock the onChange handler

    const { getByText } = render(
      <SelectField
        label="Test Select"
        options={TYPE_PRICE}
        onChange={onChangeMock}
        name="testSelect"
      />,
    );

    fireEvent.click(getByText("Select..."));
    fireEvent.click(getByText(TYPE_PRICE[2].option));

    expect(onChangeMock).toHaveBeenCalledTimes(1);
    expect(onChangeMock).toHaveBeenCalledWith(TYPE_PRICE[2].value);
  });
});
