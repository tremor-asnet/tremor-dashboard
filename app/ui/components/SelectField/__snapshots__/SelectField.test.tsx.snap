// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField Component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full"
      >
        <label
          class="text-secondary text-sm dark:text-lighter"
        >
          Currency
        </label>
        <div
          class="w-full min-w-[10rem] relative text-tremor-default"
        >
          <select
            class="h-full w-full absolute left-0 top-0 z-0 opacity-0"
            title="select-hidden"
          >
            <option
              class="hidden"
              disabled=""
              hidden=""
              value=""
            >
              Select...
            </option>
            <option
              class="hidden"
              value="1"
            >
              BTC
            </option>
            <option
              class="hidden"
              value="2"
            >
              CNY
            </option>
            <option
              class="hidden"
              value="3"
            >
              EUR
            </option>
            <option
              class="hidden"
              value="4"
            >
              GBP
            </option>
            <option
              class="hidden"
              value="5"
            >
              INR
            </option>
            <option
              class="hidden"
              value="6"
            >
              USD
            </option>
          </select>
          <div
            class="w-full min-w-[10rem] relative text-tremor-default select-custom dark:text-white dark:border-light dark:focus:border-white pt-[6px]"
            data-headlessui-state=""
          >
            <button
              aria-expanded="false"
              aria-haspopup="listbox"
              class="w-full outline-none text-left whitespace-nowrap truncate rounded-tremor-default focus:ring-2 transition duration-100 border pr-8 py-2 shadow-tremor-input focus:border-tremor-brand-subtle focus:ring-tremor-brand-muted dark:shadow-dark-tremor-input dark:focus:border-dark-tremor-brand-subtle dark:focus:ring-dark-tremor-brand-muted pl-3 bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border"
              data-headlessui-state=""
              id="headlessui-listbox-button-:r0:"
              type="button"
            >
              <span
                class="w-[90%] block truncate"
              >
                Select...
              </span>
              <span
                class="absolute inset-y-0 right-0 flex items-center mr-3"
              >
                <svg
                  class="tremor-Select-arrowDownIcon flex-none h-5 w-5 text-tremor-content-subtle dark:text-dark-tremor-content-subtle"
                  fill="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.9999 13.1714L16.9497 8.22168L18.3639 9.63589L11.9999 15.9999L5.63599 9.63589L7.0502 8.22168L11.9999 13.1714Z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full"
    >
      <label
        class="text-secondary text-sm dark:text-lighter"
      >
        Currency
      </label>
      <div
        class="w-full min-w-[10rem] relative text-tremor-default"
      >
        <select
          class="h-full w-full absolute left-0 top-0 z-0 opacity-0"
          title="select-hidden"
        >
          <option
            class="hidden"
            disabled=""
            hidden=""
            value=""
          >
            Select...
          </option>
          <option
            class="hidden"
            value="1"
          >
            BTC
          </option>
          <option
            class="hidden"
            value="2"
          >
            CNY
          </option>
          <option
            class="hidden"
            value="3"
          >
            EUR
          </option>
          <option
            class="hidden"
            value="4"
          >
            GBP
          </option>
          <option
            class="hidden"
            value="5"
          >
            INR
          </option>
          <option
            class="hidden"
            value="6"
          >
            USD
          </option>
        </select>
        <div
          class="w-full min-w-[10rem] relative text-tremor-default select-custom dark:text-white dark:border-light dark:focus:border-white pt-[6px]"
          data-headlessui-state=""
        >
          <button
            aria-expanded="false"
            aria-haspopup="listbox"
            class="w-full outline-none text-left whitespace-nowrap truncate rounded-tremor-default focus:ring-2 transition duration-100 border pr-8 py-2 shadow-tremor-input focus:border-tremor-brand-subtle focus:ring-tremor-brand-muted dark:shadow-dark-tremor-input dark:focus:border-dark-tremor-brand-subtle dark:focus:ring-dark-tremor-brand-muted pl-3 bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border"
            data-headlessui-state=""
            id="headlessui-listbox-button-:r0:"
            type="button"
          >
            <span
              class="w-[90%] block truncate"
            >
              Select...
            </span>
            <span
              class="absolute inset-y-0 right-0 flex items-center mr-3"
            >
              <svg
                class="tremor-Select-arrowDownIcon flex-none h-5 w-5 text-tremor-content-subtle dark:text-dark-tremor-content-subtle"
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.9999 13.1714L16.9497 8.22168L18.3639 9.63589L11.9999 15.9999L5.63599 9.63589L7.0502 8.22168L11.9999 13.1714Z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
